src/log/async_log.rs

```diff
@@ -537,33 +546,142 @@ impl Drop for AsyncAppender {
 }
 
 trait SyncWrite {
-    fn sync_write(&mut self, buf: &[u8]) -> io::Result<()>;
+    fn sync_write(&mut self, buf: &[u8], level: Level) -> io::Result<()>;
+    fn flush(&mut self) -> io::Result<()>;
 }
 
 impl SyncWrite for Stdout {
-    fn sync_write(&mut self, buf: &[u8]) -> io::Result<()> {
+    fn sync_write(&mut self, buf: &[u8], _level: Level) -> io::Result<()> {
+        let mut out = self.lock();
+        out.write_all(buf)
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
         let mut out = self.lock();
-        out.write_all(buf)?;
         out.flush()
     }
 }
 
 impl SyncWrite for File {
-    fn sync_write(&mut self, buf: &[u8]) -> io::Result<()> {
-        self.write_all(buf)?;
-        self.flush()
+    fn sync_write(&mut self, buf: &[u8], _level: Level) -> io::Result<()> {
+        self.write_all(buf)
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        Write::flush(self)
     }
 }
 
 impl SyncWrite for TcpStream {
-    fn sync_write(&mut self, buf: &[u8]) -> io::Result<()> {
+    fn sync_write(&mut self, buf: &[u8], _level: Level) -> io::Result<()> {
         self.write_all(buf)?;
+        SyncWrite::flush(self)
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
         self.write_all(&MSG_TERMINATOR[..])
```

Comment:
Is terminator written only after a call to flush ? What if i send multiple messages before i flush ? How will the remote end know when a msg ended and a new one began ?